addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/handler/SqueezeBoxServerHandler.java

```diff
@@ -839,14 +869,134 @@ public void updateListener(SqueezeBoxPlayerEventListener listener) {
+                else if (part.startsWith("id%3A")) {
+                    id = part.substring("id%3A".length());
+                    f = new Favorite(id);
+                    favorites.add(f);
+                }
+                // Favorite name
+                else if (part.startsWith("name%3A")) {
+                    name = decode(part.substring("name%3A".length()));
+                    if (f != null) {
+                        f.name = name;
+                    }
+                }
+                // Favorite type
+                else if (part.startsWith("type%3A")) {
+                    type = decode(part.substring("type%3A".length()));
+                    if (f != null) {
+                        f.type = type;
+                    }
+                }
+                // Favorite is audio
+                else if (part.startsWith("isaudio%3A")) {
+                    isaudio = new Boolean("1".matches(part.substring("isaudio%3A".length())));
+                    if (f != null) {
+                        f.isaudio = isaudio;
+                    }
+                }
+                // When "1", favorite is a submenu with additional favorites
+                else if (part.startsWith("hasitems%3A")) {
+                    hasitems = new Boolean("1".matches(part.substring("hasitems%3A".length())));
+                    if (f != null) {
+                        if (hasitems) {
+                            // Skip subfolders
+                            favorites.remove(f);
+                            f = null;
+                        } else {
+                            f.hasitems = hasitems;
+                        }
+                    }
+                }
+            }
+            updatePlayersFavoritesList(favorites);
+            updateChannelFavoritesList(favorites);
+        }
+
+        private void updatePlayersFavoritesList(List<Favorite> favorites) {
+            updatePlayer(new PlayerUpdateEvent() {
+                @Override
+                public void updateListener(SqueezeBoxPlayerEventListener listener) {
+                    listener.updateFavoritesListEvent(favorites);
+                }
+            });
+        }
+
+        private void updateChannelFavoritesList(List<Favorite> favorites) {
+            final Channel channel = getThing().getChannel(CHANNEL_FAVORITES_LIST);
+            if (channel == null) {
+                logger.debug("Channel {} doesn't exist. Delete & add thing to get channel.", CHANNEL_FAVORITES_LIST);
+                return;
+            }
+
+            // Get channel config parameter indicating whether name should be wrapped with double quotes
+            Boolean includeQuotes = Boolean.FALSE;
+            if (channel.getConfiguration().containsKey(CHANNEL_CONFIG_QUOTE_LIST)) {
+                includeQuotes = (Boolean) channel.getConfiguration().get(CHANNEL_CONFIG_QUOTE_LIST);
+            }
+
+            final String quote = includeQuotes.booleanValue() ? "\"" : "";
```

Comment:
I prefer to leave away the `final` here since they are not consistently used (in this binding) anyways.

Reply:
Ok